lib/internal/per_context/primordials.js

```diff
@@ -477,6 +479,51 @@ primordials.SafePromiseAll = (promises, mapFn) =>
     SafePromise.all(arrayToSafePromiseIterable(promises, mapFn)).then(a, b)
   );
 
+/**
+ * Should only be used for internal functions, this would produce similar
+ * results as `Promise.all` but without prototype pollution, and the return
+ * value is not a genuine Array but an array-like object.
+ * @param {Promise<any>[]} promises
+ * @param {(v: Promise<any>, k: number) => Promise<any>} [mapFn]
+ * @returns {Promise<any[]>}
+ */
+primordials.SafePromiseAllReturnArrayLike = (promises, mapFn) =>
+  new Promise((resolve, reject) => {
+    const { length } = promises;
```

Comment:
this wouldn't work for non-arraylike iterable inputs, but for internal usage obv that's a fine restriction - but probably worth noting in the comment above.

Reply:
I've updated the JSDoc to clarify which functions accept only `Array`s, which only accept array-like – and none accept `Iterable`s